File: android/app/src/main/java/com/soohoon/influe/data/model/SettlementDTO.kt
package com.soohoon.influe.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SettlementDTO(
    val id: String,
    @SerialName("workspace_id") val workspaceId: String,
    @SerialName("sponsorship_id") val sponsorshipId: String? = null,
    @SerialName("brand_name") val brandName: String,
    val amount: Double = 0.0,
    val fee: Double = 0.0,
    val tax: Double = 0.0,
    @SerialName("settlement_date") val settlementDate: String? = null,
    @SerialName("is_paid") val isPaid: Boolean = false,
    val memo: String = "",
    @SerialName("created_by") val createdBy: String? = null,
    @SerialName("created_at") val createdAt: String = ""
) {
    val netAmount: Double get() = amount - fee - tax
}

@Serializable
data class SettlementInsert(
    @SerialName("workspace_id") val workspaceId: String,
    @SerialName("created_by") val createdBy: String,
    @SerialName("brand_name") val brandName: String,
    val amount: Double = 0.0,
    val fee: Double = 0.0,
    val tax: Double = 0.0,
    @SerialName("settlement_date") val settlementDate: String? = null,
    @SerialName("is_paid") val isPaid: Boolean = false,
    val memo: String = "",
    @SerialName("sponsorship_id") val sponsorshipId: String? = null
)

File: CreatorNote/Supabase/DTOs/SettlementDTO.swift
import Foundation

struct SettlementDTO: Codable, Identifiable, Sendable {
    let id: UUID
    var workspaceId: UUID
    var sponsorshipId: UUID?
    var brandName: String
    var amount: Double
    var fee: Double
    var tax: Double
    var settlementDate: Date?
    var isPaid: Bool
    var memo: String
    var createdBy: UUID?
    var updatedBy: UUID?
    var createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id, amount, fee, tax, memo
        case workspaceId = "workspace_id"
        case sponsorshipId = "sponsorship_id"
        case brandName = "brand_name"
        case settlementDate = "settlement_date"
        case isPaid = "is_paid"
        case createdBy = "created_by"
        case updatedBy = "updated_by"
        case createdAt = "created_at"
    }

    var netAmount: Double {
        amount - fee - tax
    }
}
